src/account/mod.rs

```diff
@@ -17,13 +18,14 @@ use serde::{Deserialize, Serialize};
 use std::{
     collections::HashMap,
     convert::TryInto,
+    ops::Deref,
     path::PathBuf,
-    sync::{Arc, Mutex},
+    sync::{Arc, Mutex, RwLock},
 };
 
 mod sync;
 pub(crate) use sync::{repost_message, RepostAction};
-pub use sync::{AccountSynchronizer, SyncedAccount, TransferMetadata};
+pub use sync::{AccountSynchronizer, SyncedAccount};
 
 type AddressesLock = Arc<Mutex<Vec<IotaAddress>>>;
 type AccountAddressesLock = Arc<Mutex<HashMap<AccountIdentifier, AddressesLock>>>;
```

Comment:
my preference would be to name it just `Addresses` and `AccountAddresses` if there aren't already types named that way. I don't see much value in signaling that you first need to get a lock before being able to change the shared data.

Reply:
I'll refactor this on a subsequent PR.